### PR TITLE
Start to add support for Electra

### PR DESCRIPTION
**Pull request by jtraglia**
This PR makes the necessary changes (adds new structures and updates versioned structures) for Electra. Sorry, it contains a lot of new files and it lacks tests. I would like to pull in these changes into an electra branch, so that multiple people (myself included) can add tests later, in smaller PRs. After this PR gets merged, I'll make the equivalent PR for `go-builder-client`. We need to wait because `go-eth2-client` is a dependency and it currently needs a `replace` directive.

### CODE REVIEW

File: spec/phase0/attestation_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func TestAttestationJSON(t *testing.T) {
 		{
 			name:  "AggregationBitsInvalid",
 			input: []byte(`{"aggregation_bits":"invalid","data":{"slot":"100","index":"1","beacon_block_root":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f","source":{"epoch":"1","root":"0x202122232425262728292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f"},"target":{"epoch":"2","root":"0x404142434445464748494a4b4c4d4e4f505152535455565758595a5b5c5d5e5f"}},"signature":"0x606162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9fa0a1a2a3a4a5a6a7a8a9aaabacadaeafb0b1b2b3b4b5b6b7b8b9babbbcbdbebf"}`),
-			err:   "invalid value for beacon block root: encoding/hex: invalid byte: U+0069 'i'",
+			err:   "invalid value for aggregation bits: encoding/hex: invalid byte: U+0069 'i'",
```
Comment (jtraglia) [inline spec/phase0/attestation_test.go:55]: This was an incorrect test for the typo above.

File: api/v1/deneb/generate.go [hunk 2]
```diff
@@ -14,6 +14,6 @@
...
-//go:generate sszgen --include ../../../spec/phase0,../../../spec/altair,../../../spec/bellatrix,../../../spec/capella,../../../spec/deneb -path . --suffix ssz -objs BlindedBeaconBlock,BlindedBeaconBlockBody,BlindedBlobSidecar,BlindedBlockContents,BlockContents,SignedBlindedBeaconBlock,SignedBlindedBlockContents,SignedBlockContents
-//go:generate goimports -w blindedbeaconblock_ssz.go blindedbeaconblockbody_ssz.go blindedblobsidecar_ssz.go blindedblockcontents_ssz.go blockcontents_ssz.go signedblindedbeaconblock_ssz.go signedblindedblockcontents_ssz.go,signedblockcontents_ssz.go
+//go:generate rm -f blindedbeaconblock_ssz.go blindedbeaconblockbody_ssz.go blockcontents_ssz.go signedblindedbeaconblock_ssz.go signedblockcontents_ssz.go
+//go:generate sszgen --include ../../../spec/phase0,../../../spec/altair,../../../spec/bellatrix,../../../spec/capella,../../../spec/deneb -path . --suffix ssz -objs BlindedBeaconBlock,BlindedBeaconBlockBody,BlockContents,SignedBlindedBeaconBlock,SignedBlockContents
+//go:generate goimports -w blindedbeaconblock_ssz.go blindedbeaconblockbody_ssz.go blockcontents_ssz.go signedblindedbeaconblock_ssz.go signedblockcontents_ssz.go
```
Comment (jtraglia) [inline api/v1/deneb/generate.go:19]: These commands contained references to files which no longer exist. Noticed this when making the electra version.

File: spec/phase0/types.go
Comment (jtraglia) [inline spec/phase0/types.go:1]: In this file, I added `UnmarshalJSON` and `MarshalJSON` methods for `Epoch`. This simplified other unmarshaling functions.

File: spec/versionedattestation.go
Comment (jtraglia) [inline spec/versionedattestation.go:1]: This is a notable addition. I wasn't exactly sure how to handle the new `Attestation` type, so I decided to make a new `VersionedAttestation` structure like we do with other structures.

**Comment by jtraglia** [comment]
I added two `nolint:gocyclo` comments for these:

```
spec/versionedsignedbeaconblock.go:272:1: cyclomatic complexity 31 of func `(*VersionedSignedBeaconBlock).Attestations` is high (> 30) (gocyclo)
func (v *VersionedSignedBeaconBlock) Attestations() ([]VersionedAttestation, error) {
^
spec/electra/beaconblockbody_json.go:71:1: cyclomatic complexity 31 of func `(*BeaconBlockBody).UnmarshalJSON` is high (> 30) (gocyclo)
func (b *BeaconBlockBody) UnmarshalJSON(input []byte) error {
^
```

Not really sure how I would make these less complex.

**Comment by mcdee** [comment]
The Electra changes themselves look good.

There are a fair number of non-Electra changes in here.  Would it be possible to remove these from this PR, to avoid additional pain when we come to merge this branch back into master?  Anything non-cosmetic can be pushed to a separate PR to be included directly into master.  Thanks.

**Comment by jtraglia** [comment]
Yes, good suggestion. I'll make those changes first thing tomorrow morning.

**Comment by jtraglia** [comment]
@mcdee Could you fix the linter issues for me? I'm not sure why there are magic number findings now.

**Review by mcdee — APPROVED** [review]

**Pull request merged**